src/helper/UserSession.cpp

```diff
@@ -100,17 +100,18 @@ namespace SDDM {
 
     void UserSession::stop()
```

Comment:
Silly question (last one I promise) - what's the problem with calling `terminate()`?

It seems that the more notable change is that we omit `waitForFinished()` will be blocking for potentially notable amount of time.

Reply:
we do still call terminate and waitForFinished() we've just wrapped things in an if statement.

Reply:
Right, I meant - what's the problem with calling `terminate()` and friends like `waitForFinished()`) in the `state() == QProcess::NotRunning` case?

Reply:
We first call terminate which is more gentle, if terminate doesn't make the project finish, we kill it.